sbc_compassion/tools/import_letter_functions.py

```diff
@@ -21,13 +21,117 @@
 import checkboxreader as cbr
 import bluecornerfinder as bcf
 import numpy as np
+from math import ceil
 from glob import glob
 from wand.image import Image
 
+from openerp import _
+
 
 ##########################################################################
 #                           GENERAL METHODS                              #
 ##########################################################################
+def testline(env, line_vals, csv_file, doc_name):
+    """ Test each line and verify the data given by the csv.
+    If a line does not correspond to the csv, this function erases it.
+    :param env: Odoo environment
+    :param dict line_vals: Dictionnary that will be used to create line
+    :param One2many('test.import.letter.line') csv_file: CSV used to check
+    :param str doc_name: Document name
+    :returns: test (True if equal to csv), error
+    :rtype: bool, str
+    """
+    # str that will be returned
+    error = ''
+    csv_data = []
+    # creates a list containing all the csv data
+    for csv_ in csv_file:
+        if os.path.splitext(csv_.name)[1] != '.csv':
+            raise Warning(_("File '{}' is not a csv file".format(csv_.name)))
+        csv_string = base64.b64decode(csv_.datas).splitlines()
+        csv_data += read_csv(csv_string)
+    # check the letters
+    values = {}
+    test = True
+    for csv_line in csv_data:
+        if csv_line['name'] == doc_name:
+            values = csv_line
+            break
+    if values == {}:
+        raise Warning(
+            _("File '{}' not found in the csv files".format(doc_name)))
+    # first test: template
+    template_id = env.ref(
+        'sbc_compassion.'+values['template_id'])
+    tpl_detected = env['sponsorship.correspondence.template'].search(
+        [('id', '=', line_vals['template_id'])])
+    temp_test = template_id == tpl_detected
```

Comment:
I think there is no need to have `temp_test` and `test` variables. You can simply do :
```
if not template_id == tpl_detected:
    error = error + 'template,\n'
...
```